### PR TITLE
Don't deploy site.pp on Puppet >= 4.0

### PR DESCRIPTION
**Pull request by sathieu**
even if puppet#15106 was fixed in Puppet 3.3

### CODE REVIEW

**Comment by mmoll** [comment]
If it's only for that puppet bug, I'd say, we can raise the minimum version in metadata and just remove it, but I'd like to wait for other people's opinions.

**Comment by domcleal** [comment]
I'd also be OK with raising the minimum to 3.3.0 and removing it.  Otherwise this PR's also fine.

**Comment by ekohl** [comment]
I was also thinking about increasing the minimum version. I don't think there's a reason to stay on puppet < 3.3.

**Comment by mmoll** [comment]
@sathieu so, please remove it and bump the minimum Puppet version :)

**Comment by sathieu** [comment]
I've done as @mmoll said, and I've added a second commit dropping $server_manifest_path.

File: spec/classes/puppet_server_config_spec.rb [hunk 1]
```diff
@@ -136,12 +136,7 @@
             :group => nil,
             :mode => '0755',
           })
-
-          should contain_file("#{codedir}/manifests/site.pp").with({
-            :ensure  => 'file',
-            :replace => false,
-            :content => "# site.pp must exist (puppet #15106, foreman #1708)\n",
-          })
+	  end
```
Comment (mmoll) [inline spec/classes/puppet_server_config_spec.rb]: :end: :no_good: ;)

**Comment by domcleal** [comment]
:+1: tested on 3.8 and it seems to work fine.

**Comment by mmoll** [comment]
merged, merci @sathieu!

**Comment by ekohl** [comment]
This should be a major version bump, right?

**Comment by mmoll** [comment]
for the removed parameter, true.

**Pull request merged**